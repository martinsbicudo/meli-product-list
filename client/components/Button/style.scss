.Button {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  color: map-get($colors, "meli-mine-shaft");
  background-color: map-get($colors, "meli-gallery");
  padding: 1.1rem;
  font-size: 1.4rem;
  border-radius: .2rem;
  border: .1rem solid transparent;
  outline: 0;

  &:disabled > * { opacity: .5; }

  &:not([disabled]) { cursor: pointer; }

  &:focus { border-color: map-get($colors, "meli-denim"); }

  &DodgerBlue {
    color: map-get($colors, "meli-white");
    background-color: map-get($colors, "meli-dodger-blue");
  }

  &Bigger {
    padding: 2rem;
    font-size: 2.2rem;
  }
}
