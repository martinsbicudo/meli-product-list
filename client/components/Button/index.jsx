import React from 'react';
import {
  string,
  node,
  array,
  oneOfType,
} from 'prop-types';
import classnames from 'classnames';
import style from './style.scss';

const Button = ({
  className,
  type,
  modifiers,
  children,
  ...props
}) => (
  <button
    className={classnames(
      style.Button,
      className,
      {
        [style.ButtonDodgerBlue]: modifiers.includes('dodger-blue'),
        [style.ButtonBigger]: modifiers.includes('bigger'),
      },
    )}
    type={type}
    {...props}
  >
    {children}
  </button>
);

Button.propTypes = {
  className: string,
  type: string,
  modifiers: array,
  children: oneOfType([string, node]).isRequired,
};

Button.defaultProps = {
  className: '',
  type: 'button',
  modifiers: [],
};

export default Button;
