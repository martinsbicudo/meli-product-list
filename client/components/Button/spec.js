import Button from '.';

describe('Components - <Button />', () => {
  const testMessage = 'TEST_MESSAGE';
  const createWrapper = (
    message = testMessage,
    props = {},
  ) => render(
    <Button {...props}>{message}</Button>,
  );

  test('Should has test message', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessage)).toBeInTheDocument();
  });

  test('Should calls onClick', () => {
    const spyOnClick = jest.fn();
    const { getByText } = createWrapper(undefined, { onClick: spyOnClick });

    fireEvent.click(getByText(testMessage));

    expect(spyOnClick).toHaveBeenCalledTimes(1);
    spyOnClick.mockRestore();
  });

  test('Should set type Button', () => {
    const { getByText } = createWrapper(undefined, { type: 'submit' });

    expect(getByText(testMessage).type).toBe('submit');
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
