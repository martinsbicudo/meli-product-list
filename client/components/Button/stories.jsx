import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from '.';

storiesOf('Button', module)
  .add('With text', () => (
    <Button>
      With text
    </Button>
  ))
  .add('With Node/Component', () => (
    <Button>
      <strong>
        With Node/Component
      </strong>
    </Button>
  ))
  .add('With [bigger] modifier', () => (
    <Button modifiers={['bigger']}>
      With bigger modifier
    </Button>
  ))
  .add('With [dodger-blue] modifier', () => (
    <Button modifiers={['dodger-blue']}>
      With dodger-blue modifier
    </Button>
  ))
  .add('With both modifiers', () => (
    <Button modifiers={['bigger', 'dodger-blue']}>
      With both modifier
    </Button>
  ))
  .add('With [onClick] modifiers', () => (
    <Button onClick={() => console.log('onClick was called') /* eslint-disable-line */}>
      With onClick modifier
    </Button>
  ));
