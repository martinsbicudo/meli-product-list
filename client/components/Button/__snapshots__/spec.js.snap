// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - <Button /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button"
    type="button"
  >
    TEST_MESSAGE
  </button>
</DocumentFragment>
`;
